### PR TITLE
Add urls 

### PR DESCRIPTION
**Pull request by jsa34**
Adding URLs to sites

### CODE REVIEW

File: controller/front-end/sites.js [hunk 1]
```diff
@@ -90,6 +92,172 @@ function initSitesController(dashboard, router) {
 			return next(error);
 		}
 	});
+	// List of all URLs associated with site
+	router.get('/sites/:siteId/urls', requirePermission('read'), async (request, response) => {
+		response.render('template/sites/urls', {
```
Comment (joeyciechanowicz) [inline controller/front-end/sites.js]: Am I missing something or does `template/sites/urls` not exist?

File: controller/front-end/sites.js [hunk 2]
```diff
@@ -90,6 +92,172 @@ function initSitesController(dashboard, router) {
...
+			site: await Site.fetchOneById(request.params.siteId)
```
Comment (joeyciechanowicz) [inline controller/front-end/sites.js]: What happens if `Site.fetchOneById` doesn't find anything? Looking at the Bookshelf docs, the methods throw things like `Model.NotFoundError` which would then be caught by express?

File: controller/front-end/sites.js [hunk 3]
```diff
@@ -90,6 +92,172 @@ function initSitesController(dashboard, router) {
...
+			return next(error);
+		}
+	});
+
+	// Display the URL edit page
+	router.get('/sites/:siteId/urls/:urlId/edit', requirePermission('write'), async (request, response, next) => {
+		try {
+			const url = await Url.fetchOneById(request.params.urlId);
+			if (!url) {
+				return next();
```
Comment (joeyciechanowicz) [inline controller/front-end/sites.js:160]: Is the `!url` check required? Does `Url.fetchOneById` not throw an exception if nothing was found?

File: controller/front-end/sites.js [hunk 4]
```diff
@@ -90,6 +92,172 @@ function initSitesController(dashboard, router) {
...
+				form: {
+					url: url.serialize()
+				}
+			});
+		} catch (error) {
+			return next(error);
+		}
+	});
+
+	// Delete a user
```
Comment (joeyciechanowicz) [inline controller/front-end/sites.js]:
```suggestion
	// Delete a URL
```

File: view/template/sites/site.dust [hunk 5]
```diff
@@ -27,4 +27,57 @@
...
+   								<a href="/sites/{site}/urls/{id}/edit">Edit</a>
+   								<a href="/sites/{site}/urls/{id}/delete">Delete</a>
+   							</td>
+   						</tr>
+   					{/urls}
+   				</tbody>
+   			</table>
+   		</div>
+   	{:else}
+   	<p>You don't have any URLs to test for this site yet; you can <a href="{site.id}/urls/new">add one here</a>.</p>
```
Comment (joeyciechanowicz) [inline view/template/sites/site.dust]: This needs a `{?permissions.write}` check round it, as it doesn't make sense to show the URL for users without the permission

File: view/template/sites/site.dust [hunk 6]
```diff
@@ -27,4 +27,57 @@
...
+   							            {?pa11yConfig.runners}
+   							                <li>Runners: {pa11yConfig.runners}</li>
+   							            {/pa11yConfig.runners}
+   							            {?pa11yConfig.standard}
+   							                <li>Standard: {pa11yConfig.standard}</li>
+   							            {/pa11yConfig.standard}
+                                       </ul>
+                                   {/pa11yConfig}
+   							</td>
+   							<td>
```
Comment (joeyciechanowicz) [inline view/template/sites/site.dust:71]: Do these actions need a `{?permissions.write}` check around them?

File: view/partial/form/url.dust [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+			<span class="field__sublabel">
+				A human-readable name for the URL, this appears across the interface
+			</span>
+		</label>
+		<input id="url-name" type="text" name="name" value="{form.url.name}" required/>
+	</div>
+
+	<div class="field field--text">
+		<label for="url-address">
+			URL to appended
```
Comment (joeyciechanowicz) [inline view/partial/form/url.dust]: I think `Path` (or `pathname`) is the name for the additional part of the URL?

```
{
	"href": "https://github.com/pa11y/sidekick/pull/95/files#diff-1931cf9f658d9d73def2b171fc2ed1d6",
	"origin": "https://github.com",
	"protocol": "https:",
	"host": "github.com",
	"hostname": "github.com",
	"pathname": "/pa11y/sidekick/pull/95/files",
	"search": "",
	"hash": "#diff-1931cf9f658d9d73def2b171fc2ed1d6"
}
```

**Review by joeyciechanowicz — CHANGES_REQUESTED** [review]

File: controller/front-end/sites.js (hunk 1 above)
Comment (jsa34) [reply controller/front-end/sites.js]: You are absolutely right - I implemented the URLs as a separate page from the sites page, but realised that it was more intuitive to include the URLs for a site on the site's page as a table... I missed removing this from when I was going with the former strategy!

File: view/template/sites/site.dust (hunk 5 above)
Comment (jsa34) [reply view/template/sites/site.dust]: Good point!

File: view/template/sites/site.dust (hunk 6 above)
Comment (jsa34) [reply view/template/sites/site.dust:71]: Good point!

File: view/partial/form/url.dust (hunk 7 above)
Comment (jsa34) [reply view/partial/form/url.dust]: Good point - I was using the "URL" nomenclature as this is what each of the paths are being referred to, but I guess baseUrl + path = URL.   Happy to rename!

File: controller/front-end/sites.js (hunk 2 above)
Comment (jsa34) [reply controller/front-end/sites.js]: Yes - it returns a 404 as below when entering an invalid siteId param for url:
image

**Review by joeyciechanowicz — APPROVED** [review]

**Pull request merged**